dora/src/bytecode/generator.rs

```diff
@@ -693,10 +693,11 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
     ) -> Option<Register> {
         match *call_type {
             CallType::Method(_, _, _) => {
-                let obj_expr = expr.object().expect("method target required");
-                let reg = self.visit_expr(obj_expr, DataDest::Alloc);
-
-                Some(reg)
+                if expr.object().is_some() {
```

Comment:
Why did `CallType::Method` change, such that `e.object()` can be None now? I don't think this should be the case.

Reply:
For static methods this should be `CallType::Fct` instead.

Reply:
I think these are the module methods – I'm using `Method` here already in preparation of modules becoming real values/instances that can extend classes and implement traits.

Reply:
Does `ModuleName::foo()` then always pass the singleton object as an implicit argument? What about modules that are just used for namespace purposes?

Maybe it makes sense to add a new CallType, e.g. `CallType::ModuleMethod`?

Reply:
It would only be used for non-namespace modules, as namespace modules cannot extend a class or implement traits – I think introducing a new type could make sense at this stage, even if it were removed it later on again.

Reply:
How do we distinguish namespace vs non-namespace modules?

I think `CallType::ModuleMethod` makes sense even with `self` as implicit argument just like in `CallType::Method`: The difference is that with `CallType::Method` there is an explicit object argument in the AST (`<obj>.method(<args>)`) whereas with module methods there isn't (`ModuleName::method(<args>)`).

Reply:
I think in general I think it's worth thinking whether it's desirable to have free-standing `fun`s in a namespace module vs. putting them in a "real" module, especially when looking at importing them ("static imports a la Java" have been the bane of my existence for quite a while).

Regarding the explicit object arg, I'd try to unify the syntax further in the future, because in the end `ModuleName::method(<args>)` doesn't really convey the info that, like other values `<obj>.method(<args>)`, it's an instance that the methods are called upon (just that `module`s have a "static" name that is known at compile-time).

Ideally I'd prefer having both use the same dot-syntax in the future `<obj>.method(<args>)` and `<module>.method(<args>)` to better convey this information. But that should surely be discussed elsewhere.

I'll introduce `ModuleMethod` and adjust the PR.